shared/cert.go

```diff
@@ -430,19 +430,24 @@ func GenerateMemCert(client bool, options CertOptions) ([]byte, []byte, error) {
 	return cert, key, nil
 }
 
+// ParseCert parse a X.509 certificate from the given byte slice and return its parsed content.
+func ParseCert(cert []byte) (*x509.Certificate, error) {
```

Comment:
Would it be useful to adapt this to take an io.Reader instead so that a bytes.Buffer or actual file handle can be used? e.g. https://pkg.go.dev/bytes#example-Buffer-Reader

Reply:
It looks that the upstream `encoding/pem` is a limitation here. Whilst it accepts a writer for encoding, decoding is only supported using a `[]byte` slice. See https://pkg.go.dev/encoding/pem#Decode. 

So even if we accept an `io.Reader` here, we would still need to read it's contents and create a slice before we can decode the contents.